pallets/gear-program/src/lib.rs

```diff
@@ -122,16 +128,20 @@ pub mod pallet {
             Self::resume_program_impl(
                 program_id,
                 memory_pages,
+                wait_list,
                 <frame_system::Pallet<T>>::block_number().unique_saturated_into(),
             )?;
 
+            let program_account = &<T::AccountId as common::Origin>::from_origin(program_id);
             T::Currency::transfer(
                 &account,
-                &<T::AccountId as common::Origin>::from_origin(program_id),
+                program_account,
                 value,
                 ExistenceRequirement::AllowDeath,
             )?;
 
+            T::Currency::extend_lock(LOCK_ID, program_account, value, WithdrawReasons::FEE);
```

Comment:
I don't think that this kind of locking is the best way. We don't use it in any other places. Wouldn't it better to use same mech we used to - reserve?

Reply:
created corresponding issue = #762